src/Symfony/Component/Security/Core/Authorization/Voter/Vote.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Core\Authorization\Voter;
+
+class Vote
+{
+    use AccessTrait;
+
+    private $reason;
+    private $parameters;
```

Comment:
Is it like a context? 
Also where are they used?

Reply:
Yes, imagine that case :
```php
if (($age = $subject->getAge()) < 18) {
    return $this->deny('You are not adult.', ['age' => $age])
}
```

Reply:
I would call it like context in the workflow to have same naming
https://symfony.com/blog/new-in-symfony-4-3-workflow-improvements#added-a-context-to-workflow-apply